homeassistant/components/mqtt/mixins.py

```diff
@@ -223,6 +224,27 @@ def validate_device_has_at_least_one_identifier(value: ConfigType) -> ConfigType
 )
 
 
+def validate_modern_schema(domain: str) -> Callable:
+    """Warn once when a legacy platform schema is used."""
+    warned = set()
+
+    def validator(config: ConfigType) -> ConfigType:
+        """Return a validator."""
+        nonlocal warned
+
+        if domain in warned:
+            return config
+
+        _LOGGER.warning(
+            "Manual configured MQTT item found at platform key '%s'. Manual MQTT item configurations have been moved to the integration key",
+            domain,
+        )
```

Comment:
Maybe this makes it a bit more clear what needs to be done?
```suggestion
        _LOGGER.warning(
            "Manually configured MQTT item(s) found under platform key '%s'. Manually configured MQTT item configurations have been moved to the integration key, please move the configuration of MQTT %s to mqtt->%s",
            domain,
            domain,
            domain,
        )
```